scene/gui/text_edit.cpp

```diff
@@ -1364,9 +1370,9 @@ void TextEdit::_notification(int p_what) {
 #else
 					int caret_width = 1;
 #endif
-					if (!clipped && cursor.line == line && ((line_wrap_index == line_wrap_amount) || (cursor.column != TS->shaped_text_get_range(rid).y))) {
-						is_cursor_line_visible = true;
-						cursor_pos.y = line_top_offset_y;
+
+					if (!clipped && cursor.line == line && line_wrap_index == caret_wrap_index) {
```

Comment:
CC @bruvzg, I'm not entirely sure about this change. Was there any reason this was ran on all wrapped lines rather then where the caret is?